lib/platform-stats.js

```diff
@@ -165,29 +284,55 @@ export const mapParticipantsToIds = async (pgClient, participantsSet) => {
     INSERT INTO participants (participant_address)
     SELECT UNNEST($1::TEXT[]) AS participant_address
     ON CONFLICT(participant_address) DO UPDATE
-      -- this no-op update is needed to populate "RETURNING id"
+      -- this no-op update is needed to populate "RETURNING id, participant_address"
       SET participant_address = EXCLUDED.participant_address
-    RETURNING id
+    RETURNING id, participant_address
   `, [
     newAddresses
   ])
 
   assert.strictEqual(created.length, newAddresses.length)
-  return ids.concat(created.map(r => r.id))
+  // eslint-disable-next-line camelcase
+  for (const { id, participant_address } of created) {
+    participantsMap.set(participant_address, id)
+  }
+
+  return participantsMap
 }
 
 /**
  * @param {import('./typings.js').CreatePgClient} createPgClient
+ * @param {object} options
+ * @param {Array<Function>} [options.functionsToRun]
+ */
+export const refreshDatabase = async (
+  createPgClient,
+  { functionsToRun = [updateTopMeasurementParticipants, aggregateAndCleanupRecentData, updateMonthlyActiveStationCount] } = {}
+) => {
+  const pgClient = await createPgClient()
+
+  try {
+    const results = await Promise.allSettled(
+      functionsToRun.map(func => func(pgClient))
+    )
+
+    results.forEach((result, index) => {
+      if (result.status === 'rejected') {
+        console.error(`Error running function ${functionsToRun[index].name}:`, result.reason)
+      }
+    })
+  } finally {
+    await pgClient.end()
+  }
```

Comment:
```suggestion
  const results = await Promise.allSettled(
    functionsToRun.map(func => func(pgClient))
  )

  results.forEach((result, index) => {
    if (result.status === 'rejected') {
      console.error(`Error running function ${functionsToRun[index].name}:`, result.reason)
    }
  })

  await pgClient.end()
```

No `try/catch` necessary, as `Promise.allSettled` doesn't reject